lib/spack/spack/fetch_strategy.py

```diff
@@ -945,9 +944,21 @@ def source_id(self):
         return self.revision
 
     def get_source_id(self):
-        output = self.svn('info', '--xml', self.url, output=str)
-        info = xml.etree.ElementTree.fromstring(output)
-        return info.find('entry/commit').get('revision')
+        try:
+            import xml.etree.ElementTree
+        except ImportError:
+            # This python install doesn't have xml support
+            output = self.svn('info', self.url, output=str)
+            if not output:
+                return None
+            lines = output.split('\n')
+            for line in lines:
+                if line.startswith('Revision:'):
+                    return line.split()[-1]
```

Comment:
If the code above works in all cases, why don't we just use it instead of keeping the `Etree` code?

Reply:
It doesn't work in Japanese, or presumably with other locale other than English/C. See #11466

Reply:
Suggestion: replace the whole thing with:

```python
output = self.svn('info', '--xml', self.url, output=str)
match = re.search(r'<commit\s+revision\s*=\s*"(\d+)">', s, re.M) 
if not match:
    return None
return int(match.group(1))
```

This is not ideal (as it's not an actual XML parser) but I believe it works reliably, and I doubt the XML format will change substantially.

Reply:
Note that this works in both cases because it's XML output and the structure is reliable, but we don't rely on `Etree` to parse it.

Reply:
I agree it will likely work in both cases, but I think we should still keep the `Etree` code for when it's available, with this as the backstop instead of the older code. Pushing a version with that change.